saseg_runner/runner.py

```diff
@@ -1,80 +1,200 @@
+"""Module to contain the EGRunner class and the associated function (which will likely be
+deprecated in the future)
+"""
 # %%
-import win32com.client
-import subprocess
-import re
+import datetime
 import os
-from pathlib import Path
+import re
 import shutil
-from typing import Union
+import subprocess
 import sys
-import click
 import time
-import datetime
 
-SCRIPTDIR_PATH = Path(os.path.dirname(__file__)).resolve()
+from pathlib import Path
+from typing import Union
+
+import click
+import pythoncom
+import win32com.client
+
+SCRIPTDIR_PATH = Path(__file__).parent.resolve()
+DEFAULT_PROFILE_NAME = "SAS Asia"
+DEFAULT_EG_VERSION = "7.1"
 
 
-def run_egp(
-        egp_path: Union[str, Path],
-        profile_name: str = 'SAS Asia',
-        eg_version: str = '7.1',
+class SASEGRunner:
+    """Class that can be used to save settings for running multiple EG files."""
+
+    def __init__(
+        self,
+        profile_name: str = DEFAULT_PROFILE_NAME,
+        eg_version: str = DEFAULT_EG_VERSION,
         overwrite: bool = False,
         remove_log: bool = True,
         verbose: bool = False,
-        ) -> None:
-    """
-    execute egp_path
-
-    Parameters
-    ------------
-    egp_path : Union[str, Path]
-        SAS Enterprise Guide file path.
-    profile_name : str
-        profile name to use
-    overwrite: bool
-        controls whether to save the egp file after exection. if False, timestamp is added to filename. The default is False.
-    remove_log: bool
-        wether remove log files or not. the default is True.
-    verbose: bool
-        default is False
+    ):
+        """Class that can be used to save settings for running multiple EG files.
+
+        Args:
+            profile_name (str, optional): profile name to use when running. Defaults to the value
+                when the instance was created.
+            eg_version (str, optional): Which version of EG to use. Defaults to the value when
+                the instance was created.
+            overwrite (bool, optional): controls whether to save the egp file after exection.
+                if False, timestamp is added to filename. Defaults to the value when the
+                instance was created.
+            remove_log (bool, optional): Whether to remove log files or not. Defaults to the
+                value when the instance was created.
+            verbose (bool, optional): [description]. Defaults to the value when the
```

Comment:
I'm not sure what the sentence 'Defaults to the value when the instance was created' says.

Is it better to 'Defaults to True' or something?

Reply:
Yep, late night coding haha. Done!